PubSub/tests/Unit/SubscriptionTest.php

```diff
@@ -492,6 +565,83 @@ public function testModifyAckDeadlineBatch()
         $this->subscription->modifyAckDeadlineBatch($messages, $seconds, ['foo' => 'bar']);
     }
 
+    /**
+     * The method modifyAckDeadlineBatch should suppress any exception if it is
+     * with a reason with EOD.
+     */
+    public function testmodifyAckDeadlineBatchSuppressesExceptionsForEod()
+    {
+        $ackIds = [
+            'foobar',
+            'otherAckId'
+        ];
+        $seconds = 10;
+        $messages = [];
+
+        foreach ($ackIds as $id) {
+            $messages[] = new Message([], ['ackId' => $id]);
+        }
+        
+        // The JSON exception msg for a failure in a sub with EOD enabled
+        $exMsg = '{"error":{"code":400,"message":"...","status":"INVALID_ARGUMENT","details":[{"@type":"type.googleapis.com/google.rpc.ErrorInfo","reason":"EXACTLY_ONCE_ACKID_FAILURE","domain":"pubsub.googleapis.com","metadata":{"foobar":"PERMANENT_FAILURE_INVALID_ACK_ID","otherAckId":"PERMANENT_FAILURE_INVALID_ACK_ID"}},{"@type":"type.googleapis.com/google.rpc.DebugInfo","detail":"..."}]}}'; //phpcs:ignore
```

Comment:
If you like this approach, feel free to ignore this suggestion, but adding this as a separate fixture that we pull into the test might be nice.

Reply:
So, I think it's best to create a task for this and take this change in a small PR later.
I'll take care of it in a following PR.
Hope that works?